---
title: "PLATCOV Statistical Analysis: Regeneron and Remdesivir"
author: "James Watson"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    toc: yes
    fig_caption: yes
    keep_md: yes
    number_sections: yes
  pdf_document:
    toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(cache = F, cache.comments = FALSE,
                      echo = F, include = TRUE, 
                      fig.width = 8, fig.height = 8,
                      fig.pos = 'H',dev = 'png', dpi = 300)

library(rstan)
library(loo)
library(RColorBrewer)
library(reshape2)
library(plotrix)
library(brms)
```



# Preamble

This Statistical Analysis Plan (SAP) is written for the PLATCOV trial (registered at ClinicalTrials.gov: <https://clinicaltrials.gov/ct2/show/NCT05041907>).

Data preparation is done in a different R script called data_prep.R. This Markdown script assumes that the data are saved in a .csv file *interim_dat.csv* in long format. The file *interim_dat.csv* contains the patient clinical and viral load data with the following column headers:

-   ID: anonymized patient id code
-   Time: time from randomization
-   Trt: treatment allocation as written in CRF
-   Site: site at enrollment
-   Timepoint_ID: Day of study (integer 0 to 14)
-   BARCODE: unique sample id code (in the following format: PLT-site-number-swab type-timepoint)
-   Swab_ID: RTS or TSL (right versus left tonsil)
-   Plate: unique Plate ID for the PCR assay (matching with plate identifiers in interim_control_dat.csv)
-   Rand_date: date of randomization
-   Any_dose: (0/1) any doses of any manufacturer received
-   N_dose: integer number of doses received (any manufacturer)
-   Antibody_test: 0/1 (negative/positive for SARS-CoV-2 antibody rapid test)
-   Weight (kg)
-   BMI: kg/weight\^2
-   Age: (years - has to be between 18-50)
-   Sex: 0/1 (male: 1; female/other: 0)
-   Symptom_onset: time since onset of symptoms (days)
-   Variant: variant of concern (using standard WHO terminology for the main lineages, reference will be the predominant variant in the data set at the start of the study)
-   CT_NS: observed CT value for the N/S gene
-   CT_RNaseP: observed CT value for the human RNase P gene
-   Per_protocol_sample: whether at the time of sampling the patient was still in per protocol with respect to drug dosing
-   IgG: + IgG band on the LFT
-   IgM: + IgM band on the LFT
-   log10_viral_load: log10 number of viral copies per mL (estimated from control samples using a mixed effects model)
-   log10_cens_vl: censoring value

## Computational setup

```{r}
## information on software/hardware used
version
sessionInfo()

study_threshold = 1.125

rstan_options(auto_write = TRUE)
## parameters for the analysis
Dmax = 8
RUN_MODELS = F

my_probs = c(0.025, 0.1, .5, .9, .975)
source('functions.R')
```

# Overview of data
## Load data

```{r}
platcov_dat = read.csv('Paper2_analysis.csv')
platcov_dat$Rand_date = as.POSIXct(platcov_dat$Rand_date)

# make per protocol summary for all patients
PP=merge(aggregate(Timepoint_ID ~ ID+Trt,
                   platcov_dat[platcov_dat$Per_protocol_sample==1, ], max),
         aggregate(Per_protocol_sample ~ ID, platcov_dat, sum),by = 'ID')
PP$Include_mITT = PP$Timepoint_ID>=2


# We remove patients who only have undetectable virus
xx_undetectble = table(platcov_dat$ID, platcov_dat$CT_NS==40)
ids_neg = names(which(xx_undetectble[,1]==0))
writeLines(sprintf('All negative samples for id: %s', ids_neg))

# Exclude from mITT pop
PP$Include_mITT[PP$ID %in% ids_neg] = F

platcov_dat = platcov_dat[!platcov_dat$ID %in% ids_neg, ]

writeLines(sprintf('PCR database contains data from %s patients',nrow(PP)))

itt_pop = read.csv('ITT_population.csv')
ind_missing = which(!itt_pop$ID %in% PP$ID & (itt_pop$Treatment %in% c('No study drug','Regeneron')))
writeLines(sprintf('we are missing patient %s',itt_pop$ID[ind_missing]))

table(itt_pop$Treatment) 
# PLT-TH1-056 dropped out before any swabs were taken
# PLT-TH1-031 also withdrew on first day

platcov_dat = platcov_dat[platcov_dat$ID %in% PP$ID[PP$Include_mITT],]
```

## Data summaries

number of patients by analysis

```{r}
platcov_dat$Remdesivir_analysis = 
  platcov_dat$Rand_date<as.POSIXct('2022-06-11') & platcov_dat$Trt != 'Regeneron'
platcov_dat$Regeneron_analysis = 
  platcov_dat$Site != 'br003' & platcov_dat$Trt != 'Remdesivir'
ind_dup = !duplicated(platcov_dat$ID)
IDs = unique(platcov_dat$ID)

writeLines(sprintf('In the remdesivir analysis there are %s patients (%s controls and %s remdesivir)',
                   length(unique(platcov_dat$ID[platcov_dat$Remdesivir_analysis & ind_dup])),
                   length(unique(platcov_dat$ID[platcov_dat$Remdesivir_analysis & 
                                                  ind_dup & 
                                                  platcov_dat$Trt=='No study drug'])),
                   length(unique(platcov_dat$ID[platcov_dat$Remdesivir_analysis & 
                                                  ind_dup & 
                                                  platcov_dat$Trt=='Remdesivir']))
))

writeLines(sprintf('In the regeneron analysis there are %s patients (%s controls and %s remdesivir)',
                   length(unique(platcov_dat$ID[platcov_dat$Regeneron_analysis & ind_dup])),
                   length(unique(platcov_dat$ID[platcov_dat$Regeneron_analysis & 
                                                  ind_dup & 
                                                  platcov_dat$Trt=='No study drug'])),
                   length(unique(platcov_dat$ID[platcov_dat$Regeneron_analysis & 
                                                  ind_dup & 
                                                  platcov_dat$Trt=='Regeneron']))
))
```


Display the per protocol matrix

```{r pp}
writeLines('Number of patients per arm in modified intention to treat analysis')
table(PP$Trt, Include_mITT = PP$Include_mITT)

writeLines('Number of swabs per protocol per treatment')
table(PP$Trt, PP_swabs = PP$Per_protocol_sample)
```



```{r data_summaries}
writeLines(sprintf('In the mITT dataset we have %s PCR datapoints on %s patients from %s sites between %s and %s:',
                   nrow(platcov_dat),
                   length(IDs),
                   length(unique(platcov_dat$Site)),
                   min(platcov_dat$Rand_date),
                   max(platcov_dat$Rand_date)))
writeLines('N patients in mITT pop by site:')
table(platcov_dat$Site[ind_dup])

# define the major lineages for the pre-specified subgroup analyses
platcov_dat$Greek_Lineage = 'Omicron'
platcov_dat$Greek_Lineage[platcov_dat$Variant=='Delta'] = 'Delta'
platcov_dat$Greek_Lineage = 
  factor(platcov_dat$Greek_Lineage, levels = c('Delta','Omicron'))

# define the G446S subgroup analyses - post hoc
platcov_dat$G446S_subgroups = 'Omicron_notG446S'
platcov_dat$G446S_subgroups[platcov_dat$Variant=='BA.1'] = 'Omicron_G446S'
platcov_dat$G446S_subgroups[platcov_dat$Variant=='Delta'] = 'Delta'
platcov_dat$G446S_subgroups = 
  factor(platcov_dat$G446S_subgroups,
         levels = c('Delta','Omicron_G446S','Omicron_notG446S'))

table(platcov_dat$Trt[ind_dup], platcov_dat$Variant[ind_dup])
table(platcov_dat$Trt[ind_dup], platcov_dat$Greek_Lineage[ind_dup])
table(platcov_dat$Trt[ind_dup], platcov_dat$G446S_subgroups[ind_dup])

# baseline samples: taken within 6 hours of randomisation
baseline_ind = platcov_dat$Timepoint_ID==0

bvl = aggregate(log10_viral_load ~ ID + Site, 
                platcov_dat[baseline_ind, ], median)
nrow(bvl) == length(IDs)

writeLines(sprintf('In the %s patients, the geometric mean baseline (defined as samples taken within 6 hours of randomisation) viral load was %s copies per mL (95 percentile interval from %s to %s; range from %s to %s)',
                   length(IDs),
                   round(10^mean(bvl$log10_viral_load)),
                   round(10^(mean(bvl$log10_viral_load)-1.96*sd(bvl$log10_viral_load))),
                   round(10^(mean(bvl$log10_viral_load)+1.96*sd(bvl$log10_viral_load))),
                   round(min(10^bvl$log10_viral_load)),
                   round(max(10^bvl$log10_viral_load))))
```


```{r variants_time}
par(las=1, cex.lab=1.5, cex.axis=1.5, family='serif')
plot(platcov_dat$Rand_date[ind_dup], 1:length(IDs),
     col= adjustcolor(as.numeric(as.factor(platcov_dat$Variant[ind_dup])),.5),
     pch = 15-platcov_dat$Variant_Imputed[ind_dup]*15+1,
     xlab='Enrollment date', ylab='Patient ID',panel.first=grid())
legend('topleft',col = 1:length(unique(platcov_dat$Variant)),cex=1.5,
       legend = levels(as.factor(platcov_dat$Variant[ind_dup])),pch=16,inset = 0.03)
legend('left',col=1, pch=c(16,1),legend = c('Genotyped','Imputed\n(based on date)'),inset = 0.03,cex=1.3)
```


Summary table

```{r table1}
plat_summary = aggregate(log10_viral_load ~ ID+Timepoint_ID+Trt+Age+Variant+Rand_date+Weight+Sex+Symptom_onset+Site,
                         data = platcov_dat, mean)
plat_summary = dplyr::arrange(plat_summary, Site, ID, Rand_date)
write.csv(plat_summary, file = '~/Downloads/platcov_regeneron_remdesivir.csv',quote = F,row.names = F)

platcov_dat$Trt[platcov_dat$Trt=='Regeneron']='Casirivimab\nimdevimab'
print(unique(platcov_dat$Trt))
platcov_dat$Trt_code = 
  factor(platcov_dat$Trt,
         levels=c("No study drug", 'Casirivimab\nimdevimab','Remdesivir'))

ind_dup = !duplicated(platcov_dat$ID)
writeLines(sprintf('female is %s%%',round(100*mean(platcov_dat$Sex[ind_dup]==0))))

writeLines(sprintf('median age is %s (IQR: %s-%s)',median(platcov_dat$Age[ind_dup]),
                   quantile(platcov_dat$Age[ind_dup],.25),
                   quantile(platcov_dat$Age[ind_dup],.75)))

writeLines(sprintf('median time since symptom onset is %s (IQR: %s-%s)',median(platcov_dat$Symptom_onset[ind_dup]),
                   quantile(platcov_dat$Symptom_onset[ind_dup],.25),
                   quantile(platcov_dat$Symptom_onset[ind_dup],.75)))
writeLines(sprintf('At least 1 vaccine is %s%%',round(100*mean(platcov_dat$N_dose[ind_dup]>0))))

xx=aggregate(log10_viral_load~ID, data=platcov_dat[platcov_dat$Timepoint_ID==0, ],mean)
round(10^quantile(xx$log10_viral_load, probs = c(0.25, .5, .75)))

writeLines('Censored values up until day 7:')
table(platcov_dat$censor[platcov_dat$Time<8])
writeLines(sprintf('The proportion censored is %s%%',round(100*mean(platcov_dat$censor[platcov_dat$Time<8]=='left'),1)))

writeLines('Variants in mITT database:')
table(platcov_dat$Variant[ind_dup])
writeLines('Variants in mITT database by treatment:')
table(platcov_dat$Trt[ind_dup], platcov_dat$Variant[ind_dup])

xx=make_baseline_table(input_data = platcov_dat)
knitr::kable(xx, caption = sprintf('Summary of patient characteristics included in the current interim analysis (n= %s). Age: median (range); baseline viral load (log10 copies per mL: mean (range)); vaccinated: %% with any number of doses; number of vaccine doses: median (range); antibody data are from rapid tests done at screening (+ is presence of IgM or IgG band).', sum(PP$Include_mITT)))
```

## Summary data plot

This includes all patients in the mITT population with the per protocol swabs.

```{r trt_data_plot}
plot_baseline_data(platcov_dat)

trt_cols = brewer.pal(n = 8,name = 'Set2')[c(1,2,4)]
names(trt_cols) = c('No study drug', 'Remdesivir','Casirivimab\nimdevimab')

par(las=1, cex.lab=1.3, cex.axis=1.3, family='serif', mfrow=c(2,3))
plot_serial_data(xx = platcov_dat[platcov_dat$Remdesivir_analysis,],
                 trt_cols = trt_cols)
title('All variants')

plot_serial_data(xx = platcov_dat[platcov_dat$Regeneron_analysis,],                 trt_cols = trt_cols)
title('All variants')

for(vv in unique(platcov_dat$Variant[platcov_dat$Variant!='BA.4'])){
  ind = platcov_dat$Variant==vv &
    platcov_dat$Regeneron_analysis
  plot_serial_data(xx = platcov_dat[ind,],
                   trt_cols = trt_cols)
  title(vv)
}
```



# Model fitting

## Simple model using brms

```{r}
# ind_fit = platcov_dat$Timepoint_ID<14
# platcov_dat$CT_RNaseP_scaled = scale(platcov_dat$CT_RNaseP)
# platcov_dat$Epoch = as.factor(platcov_dat$Epoch)
# platcov_dat$Variant = factor(platcov_dat$Variant, 
#                              levels = c('Delta','BA.1','BA.2','BA.4','BA.5'))
# platcov_dat$Trt = factor(platcov_dat$Trt, levels = c('No study drug','Casirivimab\nimdevimab'))
# 
# if(RUN_MODELS){
#   mod_naive = brm(log10_viral_load | cens(censor) ~ Time*Trt-Trt +
#                     Time*Variant+Time*Epoch+CT_RNaseP_scaled+(1+Time|ID),
#                   data = platcov_dat[ind_fit, ], 
#                   prior = c(prior(lkj(2), class = "cor"),
#                             prior(normal(5, 5), class = Intercept),
#                             prior(normal(0,1), class='b'),
#                             prior(constant(7), class = 'nu')),
#                   family = 'student', cores = 4, chains = 4)
#   save(mod_naive, file = 'Rout/naive_mod.RData')
# } else {
#   load( file = 'Rout/naive_mod.RData')
# }
# summary(mod_naive, pars=c('Trt', 'Variant','Epoch'))
# plot(mod_naive, pars=c('Trt', 'Variant','Epoch'), N=4)
```


## Full analysis

The primary analysis consists of fitting a sequence of Bayesian hierarchical models to the mITT population (viral load data up until day 7, only patients who did not leave or switch before day 3).

There are a series of model fits, which are combinations of three models:

-   Model 1: vanilla student-t regression with left censoring at 0 and with individual random effects for slope an intercept;
-   Model 2: Linear model with additional RNaseP adjustment (this needs special treatment as we need to subtract the effect of RNaseP for the linear predictions);
-   Model 3: Non-linear model (up and then down) with RNaseP adjustment.

two sets of priors:

-   Weakly informative priors
-   Quasi flat prior (multiply SDs by 10)

and two sets of covariate adjustments:

-   basic covariate adjustment (slope and intercept):
-   Variant (main Greek lineages as defined by WHO)
-   Site
-   Full covariate adjustment (slope and intercept), adding in:
- Vaccination (number of doses)
- Age (standardized to have mean=0 and sd=1)
- Time since symptom onset (days, between 0 and 4)


## Specify priors

get priors

```{r priors}
source('priors.R')
# print(all_priors)
```

## Prepare models

We make the stan data sets.

```{r}
platcov_dat_list = list()
platcov_dat_list[[1]]=platcov_dat[platcov_dat$Remdesivir_analysis,] # Remdesivir 

platcov_dat_list[[2]]=platcov_dat[platcov_dat$Regeneron_analysis,] # all Regeneron 

platcov_dat_list[[3]]=platcov_dat[platcov_dat$Rand_date < '2022-06-11' &
                                    platcov_dat$Site!='br003',] # thai sites where could be randomised to remdesivir or regen-cov 

platcov_dat_list[[4]]=platcov_dat[platcov_dat$Regeneron_analysis,] # Regeneron Subgroup analysis based on major Greek Lineage
platcov_dat_list[[5]]=platcov_dat[platcov_dat$Regeneron_analysis,] # Regeneron Subgroup analysis based on G446S mutations
platcov_dat_list[[6]]=platcov_dat[platcov_dat$Regeneron_analysis & platcov_dat$Variant!='BA.4',] # Regeneron Subgroup analysis based on Omicron sublineages

platcov_dat_list[[7]]=platcov_dat # everything

analysis_datasets = 
  c('Remdesivir',
    'Regeneron',
    'REGN_Remdesivir_contemporaneous',
    'Subgroup_Greek',
    'Subgroup_G446S',
    'Subgroup_Sublineages',
    'Full')

trt_list = list()
trt_list[[1]] = c("No study drug", 'Remdesivir')
trt_list[[2]] = c("No study drug", 'Casirivimab\nimdevimab')
trt_list[[3]] = c("No study drug", 'Remdesivir',
                  'Casirivimab\nimdevimab')
trt_list[[4]] = c("No study drug", 'Casirivimab\nimdevimab')
trt_list[[5]] = c("No study drug", 'Casirivimab\nimdevimab')
trt_list[[6]] = c("No study drug", 'Casirivimab\nimdevimab')
trt_list[[7]] = c("No study drug", 'Remdesivir',
                  'Casirivimab\nimdevimab')

for(i in 1:length(platcov_dat_list)){
  writeLines(sprintf('In %s there are %s PCR datapoints and %s patients',
                     analysis_datasets[i],
                     nrow(platcov_dat_list[[i]]),
                     length(unique(platcov_dat_list[[i]]$ID))))
  ## re-arrange so that censored values come last
  platcov_dat_list[[i]] = 
    dplyr::arrange(platcov_dat_list[[i]],
                   log10_viral_load==log10_cens_vl)
  
  # any missing timepoints or negative timepoints replace with protocol time
  ind_change_time = is.na(platcov_dat_list[[i]]$Time) | platcov_dat_list[[i]]$Time <0
  
  platcov_dat_list[[i]]$Time[ind_change_time] =
    platcov_dat_list[[i]]$Timepoint_ID[ind_change_time]
  # Datapoints that will be used in the analysis
  ind = platcov_dat_list[[i]]$Time < Dmax & 
    platcov_dat_list[[i]]$Per_protocol_sample==1 &
    platcov_dat_list[[i]]$ID %in% PP$ID[PP$Include_mITT]
  
  platcov_dat_list[[i]]$Trt = factor(platcov_dat_list[[i]]$Trt, 
                                     levels = trt_list[[i]])
  
  platcov_dat_list[[i]] = platcov_dat_list[[i]][ind, ]
  writeLines(sprintf('Analysis dataset contains %s patients and %s datapoints',
                     length(unique(platcov_dat_list[[i]]$ID)),
                     nrow(platcov_dat_list[[i]])))
}

names(platcov_dat_list) = analysis_datasets
```


```{r make_stan_datasets}
covs_base = c('Variant','Site')
covs_full=c(covs_base, 'Age_scaled','Symptom_onset')
stan_inputs = list()

for(i in 1:length(platcov_dat_list)){
  stan_inputs[[i]] = 
    make_stan_inputs(input_data_fit = platcov_dat_list[[i]],
                     int_covs_base = covs_base,
                     int_covs_full = covs_full,
                     slope_covs_base = covs_base,
                     slope_covs_full = covs_full,
                     trt_frmla = formula('~ Trt'),
                     Dmax = Dmax)
}
names(stan_inputs) = analysis_datasets

stan_inputs$REGN_Remdesivir_contemporaneous$Trt_matrix = 
  cbind(stan_inputs$REGN_Remdesivir_contemporaneous$Trt_matrix,
        as.numeric(platcov_dat_list$REGN_Remdesivir_contemporaneous$Variant!='Delta' & 
                     platcov_dat_list$REGN_Remdesivir_contemporaneous$Trt=='Casirivimab\nimdevimab'))

stan_inputs$Full$Trt_matrix = 
  cbind(stan_inputs$Full$Trt_matrix,
        as.numeric(platcov_dat_list$Full$Variant!='Delta' & 
                     platcov_dat_list$Full$Trt=='Casirivimab\nimdevimab'))

stan_inputs[['Subgroup_Greek']] = 
  make_stan_inputs(input_data_fit = platcov_dat_list[['Subgroup_Greek']], 
                   int_covs_base = c('Variant','Site'),
                   int_covs_full = c('Variant','Site','Symptom_onset',
                                     'Age_scaled'),
                   slope_covs_base = c('Variant','Site'),
                   slope_covs_full = c('Variant','Site','Symptom_onset',
                                       'Age_scaled'),
                   trt_frmla = formula('~ Trt*Greek_Lineage'),
                   Dmax = Dmax)


stan_inputs[['Subgroup_G446S']] = 
  make_stan_inputs(input_data_fit = platcov_dat_list[['Subgroup_G446S']], 
                   int_covs_base = 'Site',
                   int_covs_full = c('Site','Symptom_onset',
                                     'Age_scaled'),
                   slope_covs_base = 'Site',
                   slope_covs_full = c('Site','Symptom_onset',
                                       'Age_scaled'),
                   trt_frmla = formula('~ Trt*G446S_subgroups'),
                   Dmax = Dmax)


stan_inputs[['Subgroup_Sublineages']] = 
  make_stan_inputs(input_data_fit = platcov_dat_list[['Subgroup_Sublineages']], 
                   int_covs_base = 'Site',
                   int_covs_full = c('Site','Symptom_onset',
                                     'Age_scaled'),
                   slope_covs_base = 'Site',
                   slope_covs_full = c('Site','Symptom_onset',
                                       'Age_scaled'),
                   trt_frmla = formula('~ Trt*Variant'),
                   Dmax = Dmax)

```

## Setup model runs

List all parameters settings for the model fitting:

```{r setup_models}
all_mods = list.files('Stan_models',full.names = TRUE,pattern = '*stan')

model_settings = rbind(expand.grid(mod = all_mods[2:3],
                                   prior = 1:2,
                                   cov_matrices = 1,
                                   dataset = analysis_datasets),
                       expand.grid(mod = all_mods[2],
                                   prior = 1,
                                   cov_matrices = 2,
                                   dataset = analysis_datasets))

model_settings$Niter = 4000
model_settings$Nwarmup = 2000
model_settings$Nthin = 8
model_settings$Nchain = 4

writeLines(sprintf('We are running all models with %s chains and %s samples for each chain, discarding %s for burn-in and thining every %s, thus giving a total of %s posterior samples per model.',
                   unique(model_settings$Nchain),
                   unique(model_settings$Niter),
                   unique(model_settings$Nwarmup),
                   unique(model_settings$Nthin), 
                   unique(model_settings$Nchain*(model_settings$Niter-model_settings$Nwarmup)/model_settings$Nthin)))


save(model_settings, 
     platcov_dat_list,
     analysis_datasets,
     stan_inputs, 
     all_priors,
     file = 'Rout/model_run_setup.RData')
```

All model fitting is run on a cluster using run_models.R (bmrc.sh provides bash commands)

Load model fits

```{r}
ff = list.files('Rout/')
ff = ff[grep(pattern = 'model_fits_',x = ff)]
if(!length(ff)==nrow(model_settings)) stop('not all outputs are ready for all model settings')
ff = paste0('Rout/',ff)
```

# Results

main models

```{r}
main_mods = model_settings$prior==1&
  model_settings$cov_matrices==1&
  model_settings$mod==all_mods[2]
```

## Remdesivir

```{r remdesivir_out}
all_res_rems = which(model_settings$dataset=='Remdesivir')
all_res_rems_main = which(model_settings$dataset=='Remdesivir' & main_mods)
effect_ests_rems=list()
for(i in all_res_rems){
  load(paste0('Rout/model_fits_',i,'.RData'))
  effect_ests_rems[[which(i==all_res_rems)]] = summary(out, pars='trt_effect',use_cache=F,probs=my_probs)$summary[,c('2.5%','10%','50%','90%','97.5%'),drop=F]
  rownames(effect_ests_rems[[which(i==all_res_rems)]])='Remdesivir'
}
par(las=1, mar=c(5,10,2,2))
plot_effect_estimates(effect_ests = effect_ests_rems,
                      plot_models = 1:length(effect_ests_rems),
                      study_threshold = study_threshold,
                      my_pch = as.numeric(model_settings$mod[all_res_rems]==all_mods[3])+15
)
legend('topright',pch=15:16,legend = c('Linear','Non-linear'))
```


## Regeneron

```{r}
all_res_regn = which(model_settings$dataset=='Regeneron')
all_res_regn_main = which(model_settings$dataset=='Regeneron' & main_mods)
effect_ests_regn=list()
for(i in all_res_regn){
  load(paste0('Rout/model_fits_',i,'.RData'))
  effect_ests_regn[[which(i==all_res_regn)]] = summary(out, pars='trt_effect',use_cache=F,probs=my_probs)$summary[,c('2.5%','10%','50%','90%','97.5%'),drop=F]
  rownames(effect_ests_regn[[which(i==all_res_regn)]])='Casirivimab\nimdevimab'
}
par(las=1, mar=c(5,10,2,2))
plot_effect_estimates(effect_ests = effect_ests_regn,
                      plot_models = 1:length(effect_ests_regn),
                      study_threshold = study_threshold,
                      my_pch = as.numeric(model_settings$mod[all_res_rems]==all_mods[3])+15
)
legend('topright',pch=15:16,legend = c('Linear','Non-linear'))
```

## Both interventions together

Up until 10 June in Thai sites (where regeneron was available)

```{r regeneron_remdesivir_contemp}
all_res = which(model_settings$dataset=='REGN_Remdesivir_contemporaneous')
all_res_main = which(model_settings$dataset=='REGN_Remdesivir_contemporaneous' & main_mods)
effect_ests_all=list()
for(i in all_res){
  load(paste0('Rout/model_fits_',i,'.RData'))
  effect_ests_all[[which(i==all_res)]] = summary(out, pars='trt_effect',use_cache=F,probs=my_probs)$summary[,c('2.5%','10%','50%','90%','97.5%'),drop=F]
  rownames(effect_ests_all[[which(i==all_res)]])= c('Remdesivir',
                                                    'Casirivimab\nimdevimab (Delta)',
                                                    'Casirivimab\nimdevimab (Omicron)')
}
par(las=1, mar=c(5,10,2,2))
plot_effect_estimates(effect_ests = effect_ests_all,
                      plot_models = 1:length(effect_ests_all),
                      study_threshold = study_threshold,
                      my_pch = as.numeric(model_settings$mod[all_res]==all_mods[3])+15
)
legend('topright',pch=15:16,legend = c('Linear','Non-linear'))
```


```{r full_analysis_contemp}
load(paste0('Rout/model_fits_',all_res_main,'.RData'))
thetas_all = extract(out,pars='trt_effect')$trt_effect

trt_remdesivir = quantile(thetas_all[,1],probs = my_probs)
trt_delta_regn = quantile(thetas_all[,2],probs = my_probs)
trt_omicron_regn = 
  quantile(rowSums(thetas_all[,2:3]),probs = my_probs)
trt_matrix = rbind(trt_remdesivir, trt_delta_regn, trt_omicron_regn)
rownames(trt_matrix) = c('Remdesivir',
                         'Casirivimab\nimdevimab (Delta)',
                         'Casirivimab\nimdevimab (Omicron)')
par(las=1, mar=c(5, 11, 2, 2), family='serif',cex.lab=1.3, cex.axis=1.3, bty='n')
plot(trt_matrix[,'50%'], 1:nrow(trt_matrix), 
     panel.first=grid(nx = NA, ny = NULL),
     xlab='Change in rate of clearance (%)', ylab='',xaxt='n',
     xlim = range(trt_matrix, na.rm = T), yaxt='n', pch=16, cex=2)

abline(v=log(c(0.8, 1, 1.25, 1.5, 2)), 
       lty='dotted',  col='lightgray')
axis(1, at = log(c(0.8, 1, 1.25, 1.5, 2)), 
     labels = c(-20, 0, 25, 50, 100))
axis(2, at = 1:nrow(trt_matrix), labels = rownames(trt_matrix), tick = F)


polygon(c(-10, log(1.125), log(1.125), -10), c(-1,-1, 10, 10),col = adjustcolor('lightgrey',.4), border = NA)
abline(v=0, lty=2, lwd=3)

points(trt_matrix[,'50%'], 1:nrow(trt_matrix),pch=16,cex=2)
for(j in 1:nrow(trt_matrix)){
  lines(trt_matrix[j, c('2.5%','97.5%')], c(j,j))
  lines(trt_matrix[j, c('10%','90%')], c(j,j), lwd=3)
}

mean(thetas_all[,1] < thetas_all[,2])
mean(thetas_all[,1] > rowSums(thetas_all[,2:3]))

plot(out,pars='slope_coefs')
```



```{r everything}
everything = which(model_settings$dataset=='Full')
everything_main = which(model_settings$dataset=='Full' & main_mods)
effect_ests_everything=list()
for(i in everything){
  load(paste0('Rout/model_fits_',i,'.RData'))
  effect_ests_everything[[which(i==everything)]] = summary(out, pars='trt_effect',use_cache=F,probs=my_probs)$summary[,c('2.5%','10%','50%','90%','97.5%'),drop=F]
  rownames(effect_ests_everything[[which(i==everything)]])=
    c('Remdesivir',
      'Casirivimab\nimdevimab (Delta)',
      'Casirivimab\nimdevimab (Omicron)')
}
par(las=1, mar=c(5,10,2,2))
plot_effect_estimates(effect_ests = effect_ests_everything,
                      plot_models = 1:length(effect_ests_everything),
                      study_threshold = study_threshold,
                      my_pch = as.numeric(model_settings$mod[everything]==all_mods[3])+15
)
legend('topright',pch=15:16,legend = c('Linear','Non-linear'))
```

```{r full_analysis}
load(paste0('Rout/model_fits_',everything_main,'.RData'))
thetas_all = extract(out,pars='trt_effect')$trt_effect

trt_remdesivir = quantile(thetas_all[,1],probs = my_probs)
trt_delta_regn = quantile(thetas_all[,2],probs = my_probs)
trt_omicron_regn = 
  quantile(rowSums(thetas_all[,2:3]),probs = my_probs)
trt_matrix = rbind(trt_remdesivir, trt_delta_regn, trt_omicron_regn)
rownames(trt_matrix) = c('Remdesivir',
                         'Casirivimab\nimdevimab (Delta)',
                         'Casirivimab\nimdevimab (Omicron)')
par(las=1, mar=c(5, 11, 2, 2), family='serif',cex.lab=1.3, cex.axis=1.3, bty='n')
plot(trt_matrix[,'50%'], 1:nrow(trt_matrix), 
     panel.first=grid(nx = NA, ny = NULL),
     xlab='Change in rate of clearance (%)', ylab='',xaxt='n',
     xlim = range(trt_matrix, na.rm = T), yaxt='n', pch=16, cex=2)

abline(v=log(c(0.8, 1, 1.25, 1.5, 2)), 
       lty='dotted',  col='lightgray')
axis(1, at = log(c(0.8, 1, 1.25, 1.5, 2)), 
     labels = c(-20, 0, 25, 50, 100))
axis(2, at = 1:nrow(trt_matrix), labels = rownames(trt_matrix), tick = F)


polygon(c(-10, log(1.125), log(1.125), -10), c(-1,-1, 10, 10),col = adjustcolor('lightgrey',.4), border = NA)
abline(v=0, lty=2, lwd=3)

points(trt_matrix[,'50%'], 1:nrow(trt_matrix),pch=16,cex=2)
for(j in 1:nrow(trt_matrix)){
  lines(trt_matrix[j, c('2.5%','97.5%')], c(j,j))
  lines(trt_matrix[j, c('10%','90%')], c(j,j), lwd=3)
}

mean(thetas_all[,1] < thetas_all[,2])
mean(thetas_all[,1] > rowSums(thetas_all[,2:3]))

plot(out,pars='slope_coefs')
```

## Regeneron subgroups
### Greek variant

```{r}
Greek_res_mod = which(model_settings$dataset=='Subgroup_Greek')
Greek_res_mod_main = which(model_settings$dataset=='Subgroup_Greek' & main_mods)
effect_ests_greek=list()
for(i in Greek_res_mod){
  load(paste0('Rout/model_fits_',i,'.RData'))
  effect_ests_greek[[which(i==Greek_res_mod)]] = summary(out, pars='trt_effect',use_cache=F,probs=my_probs)$summary[,c('2.5%','10%','50%','90%','97.5%'),drop=F]
  rownames(effect_ests_greek[[which(i==Greek_res_mod)]])=
    c('Casirivimab\nimdevimab (overall)',
      'Omicron vs Delta',
      'Casirivimab\nimdevimab:Omicron')
}
par(las=1, mar=c(5,13,2,2))
plot_effect_estimates(effect_ests = effect_ests_greek,
                      plot_models = 1:length(effect_ests_greek),
                      study_threshold = study_threshold,
                      my_pch = as.numeric(model_settings$mod[all_res_rems]==all_mods[3])+15
)
legend('topright',pch=15:16,legend = c('Linear','Non-linear'))

load(paste0('Rout/model_fits_',Greek_res_mod_main,'.RData'))
trt_names =colnames(stan_inputs$Subgroup_Greek$Trt_matrix)[-1]
thetas_greek = extract(out,pars='trt_effect')$trt_effect
colnames(thetas_greek)=trt_names
trt_delta_regn = quantile(thetas_greek[,"TrtCasirivimab\nimdevimab"],probs = my_probs)
trt_omicron_regn = 
  quantile(rowSums(thetas_greek[,c("TrtCasirivimab\nimdevimab",
                                   "TrtCasirivimab\nimdevimab:Greek_LineageOmicron")]),probs = my_probs)

```

### G446S mutation

```{r}
G446S_res_mod = which(model_settings$dataset=='Subgroup_G446S')
G446S_res_mod_main = which(model_settings$dataset=='Subgroup_G446S' & main_mods)
effect_ests_g446s=list()
for(i in G446S_res_mod){
  load(paste0('Rout/model_fits_',i,'.RData'))
  effect_ests_g446s[[which(i==G446S_res_mod)]] = summary(out, pars='trt_effect',use_cache=F,probs=my_probs)$summary[,c('mean','2.5%','10%','50%','90%','97.5%'),drop=F]
  rownames(effect_ests_g446s[[which(i==G446S_res_mod)]])=
    c('Casirivimab\nimdevimab (overall)',
      'Omicron G446S',
      'Omicron not G446S',
      'Casirivimab\nimdevimab:\nOmicron G446S',
      'Casirivimab\nimdevimab:\nnot Omicron G446S')
}
par(las=1, mar=c(5,13,2,2))
plot_effect_estimates(effect_ests = effect_ests_g446s,
                      plot_models = 1:length(effect_ests_g446s),
                      study_threshold = study_threshold,
                      my_pch = as.numeric(model_settings$mod[all_res_rems]==all_mods[3])+15
)
legend('topright',pch=15:16,legend = c('Linear','Non-linear'))


load(paste0('Rout/model_fits_',G446S_res_mod_main,'.RData'))
trt_names =colnames(stan_inputs$Subgroup_G446S$Trt_matrix)[-1]
thetas_g446s = extract(out,pars='trt_effect')$trt_effect
colnames(thetas_g446s)=trt_names
trt_delta_g446s_regn = quantile(thetas_g446s[,"TrtCasirivimab\nimdevimab"],probs = my_probs)
trt_omicron_g446s_regn = 
  quantile(rowSums(thetas_g446s[,c("TrtCasirivimab\nimdevimab",
                                   "TrtCasirivimab\nimdevimab:G446S_subgroupsOmicron_G446S")]),probs = my_probs)
trt_omicron_notg446s_regn = 
  quantile(rowSums(thetas_g446s[,c("TrtCasirivimab\nimdevimab",
                                   "TrtCasirivimab\nimdevimab:G446S_subgroupsOmicron_notG446S")]),probs = my_probs)
```


### Make main figure

```{r Main_Result}
trt_matrix = 
  rbind(effect_ests_rems[[which(all_res_rems_main==all_res_rems)]],
        effect_ests_regn[[which(all_res_regn_main==all_res_regn)]],
        trt_delta_regn, trt_omicron_regn,
        trt_delta_g446s_regn, trt_omicron_g446s_regn, trt_omicron_notg446s_regn
  )
rownames(trt_matrix) = c('All variants','All variants','Delta','Omicron',
                         'Delta','Omicron:\nG446S mutated',
                         'Omicron:\nnot G446S mutated')
par(las=1, mar=c(5, 11, 2, 13), family='serif',cex.lab=1.3, cex.axis=1.3, bty='n')
plot(trt_matrix[,'50%'], 1:nrow(trt_matrix), 
     panel.first=grid(nx = NA, ny = NULL),
     xlab='Change in rate of clearance (%)', ylab='',xaxt='n',
     xlim = range(trt_matrix, na.rm = T), yaxt='n', pch=16, cex=2)

abline(v=log(c(0.8, 1, 1.25, 1.5, 2)), 
       lty='dotted',  col='lightgray')
axis(1, at = log(c(0.8, 1, 1.25, 1.5, 2)), 
     labels = c(-20, 0, 25, 50, 100))
axis(2, at = 1:nrow(trt_matrix), labels = rownames(trt_matrix), tick = F)

axis(4, at = c(1,2,3.5,6), 
     labels = c('Remdesivir',
                'Casirivimab/imdevimab',
                'Casirivimab/imdevimab:\nPost hoc\nsubgroup analysis',
                'Casirivimab/imdevimab:\nPre-specified\nsubgroup analysis'), tick = F)

polygon(c(-10, log(1.125), log(1.125), -10), c(-1,-1, 10, 10),col = adjustcolor('lightgrey',.4), border = NA)
abline(v=0, lty=2, lwd=3)

abline(h=1.5, lwd=3,lty=1,col='grey',xpd=T)
abline(h=2.5, lwd=3,lty=1,col='grey',xpd=T)
abline(h=4.5, lwd=3,lty=1,col='grey',xpd=T)
points(trt_matrix[,'50%'], 1:nrow(trt_matrix),pch=16,cex=2)
for(j in 1:nrow(trt_matrix)){
  lines(trt_matrix[j, c('2.5%','97.5%')], c(j,j))
  lines(trt_matrix[j, c('10%','90%')], c(j,j), lwd=3)
}

```



## Covariate effects and variant effects

```{r cov_effects}

```


